generated_code/aderdg.py

```diff
@@ -210,46 +211,73 @@ def addInit(self, generator):
     generator.add('projectIniCond', self.Q['kp'] <= self.db.projectQP[self.t('kl')] * iniCond['lp'])
     generator.add('evalAtQP', dofsQP['kp'] <= self.db.evalAtQP[self.t('kl')] * self.Q['lp'])
 
-  def addLocal(self, generator):
-    volumeSum = self.Q['kp']
-    for i in range(3):
-      volumeSum += self.db.kDivM[i][self.t('kl')] * self.I['lq'] * self.starMatrix(i)['qp']
-    if self.sourceMatrix():
-      volumeSum += self.I['kq'] * self.sourceMatrix()['qp']
-    volume = (self.Q['kp'] <= volumeSum)
-    generator.add('volume', volume)
-
-    localFlux = lambda i: self.Q['kp'] <= self.Q['kp'] + self.db.rDivM[i][self.t('km')] * self.db.fMrT[i][self.t('ml')] * self.I['lq'] * self.AplusT['qp']
-    localFluxPrefetch = lambda i: self.I if i == 0 else (self.Q if i == 1 else None)
-    generator.addFamily('localFlux', simpleParameterSpace(4), localFlux, localFluxPrefetch)
-
-    localFluxNodal = lambda i: self.Q['kp'] <= self.Q['kp'] + self.db.project2nFaceTo3m[i]['kn'] * self.INodal['no'] * self.AminusT['op']
-    localFluxNodalPrefetch = lambda i: self.I if i == 0 else (self.Q if i == 1 else None)
-    generator.addFamily('localFluxNodal', simpleParameterSpace(4), localFluxNodal, localFluxNodalPrefetch)
-
-  def addNeighbor(self, generator):
-    neighbourFlux = lambda h,j,i: self.Q['kp'] <= self.Q['kp'] + self.db.rDivM[i][self.t('km')] * self.db.fP[h][self.t('mn')] * self.db.rT[j][self.t('nl')] * self.I['lq'] * self.AminusT['qp']
-    neighbourFluxPrefetch = lambda h,j,i: self.I
-    generator.addFamily('neighboringFlux', simpleParameterSpace(3,4,4), neighbourFlux, neighbourFluxPrefetch)
-
-  def addTime(self, generator):
-    qShape = (self.numberOf3DBasisFunctions(), self.numberOfQuantities())
-    dQ0 = OptionalDimTensor('dQ(0)', self.Q.optName(), self.Q.optSize(), self.Q.optPos(), qShape, alignStride=True)
-    power = Scalar('power')
-    derivatives = [dQ0]
-    generator.add('derivativeTaylorExpansion(0)', self.I['kp'] <= power * dQ0['kp'])
-    for i in range(1,self.order):
-      derivativeSum = Add()
+  def addLocal(self, generator, platforms):
+    for platform in platforms:
+      name_prefix = generate_kernename_prefix(platform)
```

Comment:
Typo in method name

Reply:
Not sure if I like that pattern though -> it leads to a lot of duplicate code. Don't have a better idea right now...